### PR TITLE
Add to BM forward folding: DW-conv1x1

### PR DESCRIPTION
**Pull request by elad-c**

### CODE REVIEW

File: model_compression_toolkit/core/common/substitutions/batchnorm_folding.py [hunk 1]
```diff
@@ -182,6 +183,7 @@ def __init__(self,
                                                       with the batch normalization weights
             get_kernel_hw_fn: Function for getting the kernel height & width shape
             is_group_conv_fn: Function for checking if the linear layer is a group-convolution
+            is_bn_node_valid_fn: Function for checking whether the node to forward fold is a valid node and it's type
```
Comment (Idan-BenAmi) [inline model_compression_toolkit/core/common/substitutions/batchnorm_folding.py:186]: Not critical: consider another name: "get_bn_node_type_fn" which returns "bn", "dw" and None...

**Review by Idan-BenAmi — APPROVED** [review]

File: model_compression_toolkit/core/common/substitutions/batchnorm_folding.py (hunk 1 above)
Comment (lior-dikstein) [inline model_compression_toolkit/core/common/substitutions/batchnorm_folding.py:186]: this is better: "is_valid_bn_node_fn". especially when you init it in "is_valid_dw_conv_fn".  why do you need the word "valid"? why not just "is_bn_node_fn"?

File: model_compression_toolkit/core/common/substitutions/batchnorm_folding.py [hunk 2]
```diff
@@ -196,6 +198,7 @@ def __init__(self,
         self.update_weights_for_bn_forward_folding_fn = update_weights_for_bn_forward_folding_fn
         self.get_kernel_hw_fn = get_kernel_hw_fn
         self.is_group_conv_fn = is_group_conv_fn
+        self.is_valid_dw_conv_fn = is_bn_node_valid_fn
```
Comment (lior-dikstein) [inline model_compression_toolkit/core/common/substitutions/batchnorm_folding.py:201]: the naming is not clear? why not use the same name?

File: model_compression_toolkit/core/common/substitutions/batchnorm_folding.py [hunk 3]
```diff
@@ -163,6 +163,7 @@ def __init__(self,
                  update_weights_for_bn_forward_folding_fn: Callable,
```
Comment (lior-dikstein) [inline model_compression_toolkit/core/common/substitutions/batchnorm_folding.py:163]: Please mention that this is also for the case of DW conv

File: model_compression_toolkit/core/common/substitutions/batchnorm_folding.py [hunk 4]
```diff
@@ -227,22 +230,32 @@ def substitute(self,
...
+        bias = conv_node.get_weights_by_keys(self.bias_str)
         if not np.all(np.array(self.get_kernel_hw_fn(kernel)) == 1):
             return graph
-        bias = conv_node.get_weights_by_keys(self.bias_str)
-        gamma = bn_node.get_weights_by_keys(self.gamma_str)
-        beta = bn_node.get_weights_by_keys(self.beta_str)
-        moving_mean = bn_node.get_weights_by_keys(self.moving_mean_str)
-        moving_variance = bn_node.get_weights_by_keys(self.moving__variance_str)
-        eps = bn_node.framework_attr[self.epsilon_str]
+        is_bn, is_dw_valid = self.is_valid_dw_conv_fn(bn_node)
```
Comment (lior-dikstein) [inline model_compression_toolkit/core/common/substitutions/batchnorm_folding.py:244]: "is_dw_1x1" is better name

File: model_compression_toolkit/core/keras/graph_substitutions/substitutions/batchnorm_folding.py [hunk 5]
```diff
@@ -49,7 +49,7 @@ def batchnorm_forward_folding_node_matchers() -> [BaseNode, BaseNode]:
     Returns:
         Matcher for batch norm nodes, and source nodes.
     """
-    bn_node = NodeOperationMatcher(BatchNormalization)
+    bn_node = NodeOperationMatcher(BatchNormalization) | NodeOperationMatcher(DepthwiseConv2D)
```
Comment (lior-dikstein) [inline model_compression_toolkit/core/keras/graph_substitutions/substitutions/batchnorm_folding.py:52]: find a better name: "bn_or_dw1x1_node"

File: model_compression_toolkit/core/keras/graph_substitutions/substitutions/batchnorm_folding.py [hunk 6]
```diff
@@ -141,6 +141,23 @@ def is_group_conv_fn(node: BaseNode) -> bool:
     return (node.type == Conv2D) and node.framework_attr[GROUPS] > 1
 
 
+def is_bn_node_valid_fn(node: BaseNode) -> [bool, bool]:
```
Comment (lior-dikstein) [inline model_compression_toolkit/core/keras/graph_substitutions/substitutions/batchnorm_folding.py:144]: name, same comment as before...

**Review by lior-dikstein — APPROVED** [review]
